src/par_iter/internal.rs

```diff
@@ -90,28 +166,35 @@ pub fn bridge<PAR_ITER,C>(mut par_iter: PAR_ITER,
         fn callback<P>(mut self, mut producer: P) -> C::Result
             where P: Producer<Item=ITEM>
         {
-            let producer_cost = producer.cost(self.len);
-            let cost = self.consumer.cost(producer_cost);
-            bridge_producer_consumer(self.len, cost, producer, self.consumer)
+            if producer.weighted() || self.consumer.weighted() {
```

Comment:
Well, I get how you might acquire hidden weights - I meant that I don't know how we could actually implement a mixed policy.  We could probably add something to explicitly request the adaptive algorithm, to avoid surprise weights, but I need to think how to make that work...